### PR TITLE
Feat: check cap at deposit

### PR DESCRIPTION
**Pull request by haythem96**
# Task: check cap at deposit

**Should be merged after #712**

## Description

check cap based on ETH deposited in Euler

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Document update

## How Has This Been Tested

Please describe how to test to verify the changes. Provide instructions so we can reproduce.

## FE Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] Added video recordings if it is a UI change

## User Facing Checklist

- [ ] I fully understand the user problem this PR is solving
- [ ] I know who the target user is for this PR and have a deep understanding of that user
- [ ] I have tried this flow thinking from the pov of the target user for this PR
- [ ] I (or working w someone on team) have scheduled a user test for this PR (if it is a large change)

### CODE REVIEW

File: packages/bull-vault/src/LeverageBull.sol [hunk 1]
```diff
@@ -124,7 +133,7 @@ contract LeverageBull {
         _depositEthInEuler(ethToLend, true);
         _borrowUsdcFromEuler(usdcToBorrow);
 
-        return (ethToLend, usdcToBorrow);
+        return (ethToLend, usdcToBorrow, IEulerEToken(eToken).balanceOfUnderlying(address(this)));
```
Comment (haythem96) [inline packages/bull-vault/src/LeverageBull.sol:136]: Could be there an issue of using this balanceOf()? either here or for calc wethToLend overall? e.g if someone transfer some amount right before leverage or full rebalance ?

File: packages/bull-vault/test/integration-test/BullStrategyTestFork.t.sol [hunk 2]
```diff
@@ -275,6 +298,10 @@ contract BullStrategyTestFork is Test {
             usdcToBorrow
         );
         assertEq(IERC20(usdc).balanceOf(user1).sub(userUsdcBalanceBefore), usdcToBorrow);
+        assertTrue(
```
Comment (aleone) [inline packages/bull-vault/test/integration-test/BullStrategyTestFork.t.sol:301]: if there ever a fuzz scenario that hits this assertion?  or is the cap too high relative to the crab balance user1 deposits in?

Comment (aleone) [reply packages/bull-vault/test/integration-test/BullStrategyTestFork.t.sol:301]: Can we test an explicit revert for the cap being hit?  Doesn't have to be in a fuzz test.

**Review by aleone — APPROVED** [review]
small comments, generally lgtm

think we should test the cap revert happening on deposit either through fuzz or explicitly in a test with params set to fail

File: packages/bull-vault/src/LeverageBull.sol [hunk 3]
```diff
@@ -55,7 +57,12 @@ contract LeverageBull {
      * @param _eulerMarkets euler markets module address
      * @param _powerTokenController wPowerPerp controller address
      */
-    constructor(address _euler, address _eulerMarkets, address _powerTokenController) {
+    constructor(
+        address _owner,
+        address _euler,
+        address _eulerMarkets,
+        address _powerTokenController
+    ) Ownable() {
```
[the remark this answers is not in the document]
Comment (haythem96) [reply packages/bull-vault/src/LeverageBull.sol:65]: Later we will need to have some functions in LeverageBull with onlyOwner modifier, like ones that will be called from Auction...etc

File: packages/bull-vault/src/LeverageBull.sol [hunk 4]
```diff
@@ -105,7 +114,7 @@ contract LeverageBull {
      * @param _ethInCrab eth in crab strategy
      * @param _squeethInCrab oSQTH debt of crab strategy
      * @param _crabTotalSupply total supply of crab tokens
-     * @return ETH deposited as collateral in Euler and borrowed amount of USDC
+     * @return ETH deposited as collateral in Euler and borrowed amount of USDC, and total ETH deposited as collateral in Euler
```
Comment (alpinechicken) [inline packages/bull-vault/src/LeverageBull.sol:117]: ETH deposited as collateral in Euler, and borrowed amount of USD, and total ETH deposited as collateral in Euler

File: packages/bull-vault/src/LeverageBull.sol [hunk 5]
```diff
@@ -55,7 +57,12 @@ contract LeverageBull {
      * @param _eulerMarkets euler markets module address
      * @param _powerTokenController wPowerPerp controller address
      */
```
Comment (alpinechicken) [inline packages/bull-vault/src/LeverageBull.sol:59]: add natspec for owner

**Review by alpinechicken — APPROVED** [review]
lgtm

File: packages/bull-vault/src/LeverageBull.sol (hunk 5 above)
Comment (haythem96) [reply packages/bull-vault/src/LeverageBull.sol:60]: added

**Comment by haythem96** [comment]
> small comments, generally lgtm
> 
> think we should test the cap revert happening on deposit either through fuzz or explicitly in a test with params set to fail

Added

**Pull request merged**